packages/lisk-transactions/src/transactions/3_vote_transaction.ts

```diff
@@ -0,0 +1,483 @@
+		this.asset = tx.asset as VoteAsset;
+	}
+
+	public static create(input: CastVoteInput): object {
+		validateInputs(input);
+		const { passphrase, secondPassphrase, votes = [], unvotes = [] } = input;
+
+		const plusPrependedVotes = prependPlusToPublicKeys(votes);
+		const minusPrependedUnvotes = prependMinusToPublicKeys(unvotes);
+		const allVotes: ReadonlyArray<string> = [
+			...plusPrependedVotes,
+			...minusPrependedUnvotes,
+		];
+
+		const transaction = {
+			...createBaseTransaction(input),
+			type: 3,
+			fee: VOTE_FEE.toString(),
+			asset: {
+				votes: allVotes,
+			},
+		};
+
+		if (!passphrase) {
+			return transaction;
+		}
+
+		const {
+			address: senderId,
+			publicKey: senderPublicKey,
+		} = getAddressAndPublicKeyFromPassphrase(passphrase);
+		const transactionWithSenderInfo = {
+			...transaction,
+			senderId,
+			senderPublicKey,
+			recipientId: senderId,
+		};
+
+		const voteTransaction = new VoteTransaction(transactionWithSenderInfo);
+		voteTransaction.sign(passphrase, secondPassphrase);
+
+		return voteTransaction.toJSON();
+	}
+
+	public static fromJSON(tx: TransactionJSON): VoteTransaction {
+		const transaction = new VoteTransaction(tx);
+		const { errors, status } = transaction.validateSchema();
+
+		if (status === Status.FAIL && errors.length !== 0) {
+			throw new TransactionMultiError(
+				'Failed to validate schema.',
+				tx.id,
+				errors,
+			);
+		}
+
+		return transaction;
+	}
+
+	protected getAssetBytes(): Buffer {
+		return Buffer.from(this.asset.votes.join(''), 'utf8');
+	}
+
+	public assetToJSON(): object {
+		return {
+			votes: this.asset.votes,
+		};
+	}
+
+	public getRequiredAttributes(): Attributes {
+		const attr = super.getRequiredAttributes();
+		const publicKey = this.asset.votes.map(pkWithAction =>
+			pkWithAction.slice(1),
+		);
+
+		return {
+			[ENTITY_ACCOUNT]: {
+				...attr[ENTITY_ACCOUNT],
+				publicKey,
+			},
+		};
+	}
+
+	public verifyAgainstOtherTransactions(
+		transactions: ReadonlyArray<TransactionJSON>,
+	): TransactionResponse {
+		const sameTypeTransactions = transactions
+			.filter(
+				tx =>
+					tx.senderPublicKey === this.senderPublicKey && tx.type === this.type,
+			)
+			.map(tx => new VoteTransaction(tx));
+		const publicKeys = this.asset.votes.map(vote => vote.substring(1));
+
+		const errors = sameTypeTransactions.reduce(
+			(previous, tx) => {
+				if (
+					tx.asset.votes
+						.map(vote => vote.substring(1))
+						.some(publicKey => publicKeys.includes(publicKey))
+				) {
+					return [
+						...previous,
+						new TransactionError(
+							'Transaction includes conflicting vote',
+							tx.id,
+							'.asset.votes',
+						),
+					];
+				}
+
+				return previous;
+			},
+			[] as ReadonlyArray<TransactionError>,
+		);
+
+		return {
+			id: this.id,
+			errors,
+			status: errors.length === 0 ? Status.OK : Status.FAIL,
+		};
+	}
+
+	public processRequiredState(state: EntityMap): RequiredVoteState {
+		const accounts = state[ENTITY_ACCOUNT];
```

Comment:
Why aren't we calling `super` here?

Reply:
we don't have super here `processRequiredState` is abstract, maybe we should add to BaseTx